src/main/java/org/gradle/profiler/asyncprofiler/AsyncProfilerController.java

```diff
@@ -71,10 +74,42 @@ public void captureSnapshot(String pid) {
     public void stopSession() {
         validateStacks();
         GradleScenarioDefinition scenario = scenarioSettings.getScenario();
-        if (flamegraphGenerator.checkInstallation()) {
-            String outputBaseName = scenario.getProfileName() + "-" + profilerConfig.getEvent();
-            generateFlames(scenario.getOutputDir(), outputBaseName + "-raw", rawFlamegraphSanitizer);
-            generateFlames(scenario.getOutputDir(), outputBaseName + "-simplified", simplifiedFlamegraphSanitizer);
+        List<JfrFlameGraphGenerator.Stacks> stacks = generateStacks(scenario.getOutputDir(), scenario.getProfileName(), this.stacks);
+        new JfrFlameGraphGenerator().generateGraphs(scenario.getOutputDir(), stacks);
+    }
+
+    private List<JfrFlameGraphGenerator.Stacks> generateStacks(File outputDir, String outputBaseName, File stacksFile) {
+        JfrToStacksConverter.EventType jfrEventType = convertAsyncEventToJfrEvent(profilerConfig.getEvent());
+        List<JfrFlameGraphGenerator.Stacks> collectedStacks = new ArrayList<>();
+        collectedStacks.add(sanitizeStacks(outputDir, outputBaseName, stacksFile, jfrEventType, JfrFlameGraphGenerator.DetailLevel.RAW, rawFlamegraphSanitizer));
+        collectedStacks.add(sanitizeStacks(outputDir, outputBaseName, stacksFile, jfrEventType, JfrFlameGraphGenerator.DetailLevel.SIMPLIFIED, simplifiedFlamegraphSanitizer));
+
+        return collectedStacks;
+    }
+
+    private JfrFlameGraphGenerator.Stacks sanitizeStacks(File outputDir, String outputBaseName, File stacksFileToConvert, JfrToStacksConverter.EventType jfrEventType, JfrFlameGraphGenerator.DetailLevel level, FlameGraphSanitizer flamegraphSanitizer) {
+        String eventFileBaseName = getEventFileBaseName(outputBaseName, jfrEventType, level);
+        File sanitizedStacksFile = new File(outputDir, eventFileBaseName + "-stacks.txt");
+        flamegraphSanitizer.sanitize(stacksFileToConvert, sanitizedStacksFile);
+        return new JfrFlameGraphGenerator.Stacks(sanitizedStacksFile, jfrEventType, level, eventFileBaseName);
+    }
+
+    private String getEventFileBaseName(String outputBaseName, JfrToStacksConverter.EventType type, JfrFlameGraphGenerator.DetailLevel level) {
```

Comment:
This is a personal style but I tend to dislike `getXXX` methods which are not getters. What about `eventFileBaseNameFor(...)`

Reply:
Renamed.